src/ImageSharp/Formats/Jpg/JpegEncoderCore.cs

```diff
@@ -834,83 +833,107 @@ private void WriteStartOfScan<TColor>(PixelAccessor<TColor> pixels)
-            where TColor : struct, IPackedPixel, IEquatable<TColor>
+        private struct BlockQuad
         {
-            Block b = Block.Create();
-            Block cb = Block.Create();
-            Block cr = Block.Create();
-
-            // ReSharper disable once InconsistentNaming
-            int prevDCY = 0, prevDCCb = 0, prevDCCr = 0;
-
-            for (int y = 0; y < pixels.Height; y += 8)
-            {
-                for (int x = 0; x < pixels.Width; x += 8)
-                {
-                    this.ToYCbCr(pixels, x, y, ref b, ref cb, ref cr);
-                    prevDCY = this.WriteBlock(ref b, QuantIndex.Luminance, prevDCY);
-                    prevDCCb = this.WriteBlock(ref cb, QuantIndex.Chrominance, prevDCCb);
-                    prevDCCr = this.WriteBlock(ref cr, QuantIndex.Chrominance, prevDCCr);
-                }
-            }
-
-            b.Dispose();
-            cb.Dispose();
-            cr.Dispose();
+            public fixed float Data[4 * Block8x8F.ScalarCount];
         }
 
         /// <summary>
         /// Encodes the image with subsampling. The Cb and Cr components are each subsampled
-        /// at a factor of 2 both horizontally and vertically.
+        ///     at a factor of 2 both horizontally and vertically.
         /// </summary>
-        /// <typeparam name="TColor">The pixel format.</typeparam>
-        /// <param name="pixels">The pixel accessor providing access to the image pixels.</param>
+        /// <typeparam name="TColor">
+        /// The pixel format.
+        /// </typeparam>
+        /// <param name="pixels">
+        /// The pixel accessor providing access to the image pixels.
+        /// </param>
         private void Encode420<TColor>(PixelAccessor<TColor> pixels)
             where TColor : struct, IPackedPixel, IEquatable<TColor>
         {
-            Block b = Block.Create();
-            Block[] cb = Block.CreateArray(4);
-            Block[] cr = Block.CreateArray(4);
+            // TODO: Need a JpegEncoderScanProcessor<TColor> struct to encapsulate all this mess:
+            Block8x8F b = default(Block8x8F);
+
+            BlockQuad cb = default(BlockQuad);
+            BlockQuad cr = default(BlockQuad);
+            Block8x8F* cbPtr = (Block8x8F*)cb.Data;
+            Block8x8F* crPtr = (Block8x8F*)cr.Data;
+
+            Block8x8F temp1 = default(Block8x8F);
+            Block8x8F temp2 = default(Block8x8F);
+
+            Block8x8F onStackLuminanceQuantTable = this.luminanceQuantTable;
+            Block8x8F onStackChrominanceQuantTable = this.chrominanceQuantTable;
+
+            UnzigData unzig = UnzigData.Create();
 
             // ReSharper disable once InconsistentNaming
-            int prevDCY = 0, prevDCCb = 0, prevDCCr = 0;
+            float prevDCY = 0, prevDCCb = 0, prevDCCr = 0;
 
-            for (int y = 0; y < pixels.Height; y += 16)
+            using (var rgbBytes = new PixelArea<TColor>(8, 8, ComponentOrder.XYZ, true))
```

Comment:
Dat controversial var. With simple types I dont like it either. With complex types ... I will refactor it for consistency :)
Sometimes I introduce vars unintentionally with R#